test/jdk/java/nio/charset/StandardCharsets/Standard.java

```diff
@@ -24,112 +24,123 @@
+    public void nameMethodTest(Charset charset, String charString) {
+        assertEquals(charset.name(), charString);
+    }
 
-        check(Charset.forName("US-ASCII") == StandardCharsets.US_ASCII);
-        check(Charset.forName("ISO-8859-1") == StandardCharsets.ISO_8859_1);
-        check(Charset.forName("UTF-8") == StandardCharsets.UTF_8);
-        check(Charset.forName("UTF-16BE") == StandardCharsets.UTF_16BE);
-        check(Charset.forName("UTF-16LE") == StandardCharsets.UTF_16LE);
-        check(Charset.forName("UTF-16") == StandardCharsets.UTF_16);
-        check(Charset.forName("UTF-32BE") == StandardCharsets.UTF_32BE);
-        check(Charset.forName("UTF-32LE") == StandardCharsets.UTF_32LE);
-        check(Charset.forName("UTF-32") == StandardCharsets.UTF_32);
+    /**
+     * Validates that calling Charset.forName() on a String is equal
+     * to the matching Charset constant from the data provider.
+     */
+    @ParameterizedTest
+    @MethodSource("charsetProvider")
+    public void forNameMethodTest(Charset charset, String charString) {
+        assertEquals(Charset.forName(charString), charset);
+    }
 
-        Set<String> charsets = new HashSet<>();
-        Field standardCharsetFields[] = StandardCharsets.class.getFields();
+    /**
+     * Validates the qualities of a StandardCharsets field are as expected:
+     * The field is final, static, public, and one can access
+     * the underlying value of the field.
+     */
+    @ParameterizedTest
+    @MethodSource("charsetFields")
+    public void charsetModifiersTest(Field charsetField) throws IllegalAccessException {
+        // Check modifiers
+        assertEquals(StandardCharsets.class, charsetField.getDeclaringClass());
+        assertTrue(Modifier.isFinal(charsetField.getModifiers()));
+        assertTrue(Modifier.isStatic(charsetField.getModifiers()));
+        assertTrue(Modifier.isPublic(charsetField.getModifiers()));
+        // Check that the value can be accessed, and it is a Charset
+        Object valueOfField = charsetField.get(null);
+        assertTrue(valueOfField instanceof Charset);
+    }
 
-        for(Field charsetField : standardCharsetFields) {
-            check(StandardCharsets.class == charsetField.getDeclaringClass());
-            check(Modifier.isFinal(charsetField.getModifiers()));
-            check(Modifier.isStatic(charsetField.getModifiers()));
-            check(Modifier.isPublic(charsetField.getModifiers()));
-            Object value;
+    /**
+     * Validates that the Charsets contained in StandardCharsets are equal
+     * to the expected Charsets list defined in the test. This test should fail if
+     * either the actual or expected (standard) Charsets are modified, and
+     * the others are not.
+     */
+    @Test
+    public void correctCharsetsTest() {
+        // Grab the value from each Standard Charset field
+        List<String> actualCharsets = Arrays.stream(StandardCharsets.class.getFields()).map(field -> {
```

Comment:
Can `charsetFields()` be used here?

Reply:
Good point, changed so that the fields are reused, and `correctCharsetsTest()` calls `charsetFields()`